lib/server.js

```diff
@@ -44,196 +39,200 @@ module.exports = Server
+        name: 'ponos'
+      })
+
+      this.hermes = hermes.hermesSingletonFactory(this.opts)
+    }
 
-/**
- * Helper function that subscribes to a given queue name.
- * @private
- * @param {string} queueName Name of the queue.
- */
-Server.prototype._subscribe = function (queueName) {
-  debug('_subscribe(' + queueName + ')')
-  this.log.trace('ponos.Server: subscribing to ' + queueName)
-  if (!this._tasks[queueName]) {
-    this.log.warn({
-      queueName: queueName
-    }, 'ponos.Server: handler not defined')
-    return
+    this.hermes = Promise.promisifyAll(this.hermes)
   }
-  this.hermes.subscribe(queueName, function (job, done) {
-    this._runWorker(queueName, job, done)
-  }.bind(this))
-}
 
-/**
- * Helper function to subscribe to all queues.
- * @private
- * @return {promise} Resolved when queues are all subscribed.
- */
-Server.prototype._subscribeAll = function () {
-  debug('_subscribeAll')
-  return Promise.resolve(this.hermes.getQueues()).bind(this)
-    .map(this._subscribe)
-}
+  /**
+   * Helper function that subscribes to a given queue name.
+   * @private
+   * @param {string} queueName Name of the queue.
+   */
+  _subscribe (queueName) {
+    debug('_subscribe(' + queueName + ')')
+    this.log.trace('ponos.Server: subscribing to ' + queueName)
+    if (!this._tasks[queueName]) {
+      this.log.warn({
+        queueName: queueName
+      }, 'ponos.Server: handler not defined')
+      return
+    }
+    this.hermes.subscribe(queueName, (job, done) => {
+      this._runWorker(queueName, job, done)
+    })
+  }
 
-/**
- * Helper function that unsubscribes to a given queue name.
- * @private
- * @param {string} queueName Name of the queue.
- * @return {promise} A promise that resolves post worker creation
- *   and subscription.
- */
-Server.prototype._unsubscribe = function (queueName) {
-  debug('_unsubscribe(' + queueName + ')')
-  this.log.trace('ponos.Server: unsubscribing from ' + queueName)
-  var handler = null // null will remove all handlers
-  return this.hermes.unsubscribeAsync(queueName, handler)
-}
+  /**
+   * Helper function to subscribe to all queues.
+   * @private
+   * @return {promise} Resolved when queues are all subscribed.
+   */
+  _subscribeAll () {
+    debug('_subscribeAll')
+    return Promise.resolve(this.hermes.getQueues()).bind(this)
+      .map(this._subscribe)
+  }
 
-/**
- * Helper function to unsubscribe from all queues.
- * @private
- * @return {promise} Resolved when queues are all subscribed.
- */
-Server.prototype._unsubscribeAll = function () {
-  debug('_unsubscribeAll')
-  return Promise.resolve(this.hermes.getQueues()).bind(this)
-    .map(this._unsubscribe)
-}
+  /**
+   * Helper function that unsubscribes to a given queue name.
+   * @private
+   * @param {string} queueName Name of the queue.
+   * @return {promise} A promise that resolves post worker creation
+   *   and subscription.
+   */
+  _unsubscribe (queueName) {
+    debug('_unsubscribe(' + queueName + ')')
```

Comment:
I am beginning to wonder if debug is useful along side `log.trace`. It might just be cleaner to use logging in all instances at trace level. Thoughts?